finance-port.patch

```diff
@@ -0,0 +1,102 @@
++        SugarCompatibleActivity.__init__(self, handle)
+         self.set_title(_("Finance"))
+         self.max_participants = 1
+ 
+@@ -904,29 +906,13 @@ class Finance(activity.Activity):
+             fd.close()
+ 
+     def __save_image_cb(self, widget):
+-        image_file = tempfile.NamedTemporaryFile(mode='w+b', suffix='.png')
+-        journal_entry = datastore.create()
+-        journal_entry.metadata['title'] = self.chart.title
+-        journal_entry.metadata['keep'] = '0'
+-        journal_entry.metadata['mime_type'] = 'image/png'
+-
+-        # generate the image
++        image_file = tempfile.NamedTemporaryFile(mode='w+b', suffix='.png', dir=activity.get_activity_root())
+         self.chart.generate_image(image_file.file, 800, 600)
+         image_file.file.close()
+-        journal_entry.file_path = image_file.name
+-
+-        # generate the preview
+-        preview_str = io.StringIO()
+-        self.chart.generate_image(preview_str, activity.PREVIEW_SIZE[0],
+-                                  activity.PREVIEW_SIZE[1])
+-        journal_entry.metadata['preview'] = dbus.ByteArray(
+-            preview_str.getvalue())
+-
+-        logging.error('Create %s image file', image_file.name)
+-        datastore.write(journal_entry)
+-        self._show_journal_alert(
+-            _('Chart created'), _('Open in the Journal'),
+-            journal_entry.object_id)
++        chooser = DesktopSaveChooser(self, filename=image_file.name)
++        filename = chooser.get_filename()
++        if filename:
++            shutil.copyfile(image_file.name,filename)
+ 
+     def _show_journal_alert(self, title, msg, object_id):
+         open_alert = Alert()
+@@ -1029,19 +1015,14 @@ class Finance(activity.Activity):
+         logging.debug('chart_data %s', chart_params)
+ 
+         # save to the journal
+-        data_file = tempfile.NamedTemporaryFile(mode='w+b', suffix='.json')
+-        journal_entry = datastore.create()
+-        journal_entry.metadata['title'] = title
+-        journal_entry.metadata['keep'] = '0'
+-        journal_entry.metadata['mime_type'] = 'application/x-chart-activity'
+-        journal_entry.metadata['activity'] = 'org.sugarlabs.SimpleGraph'
+-
++        
++        data_file = tempfile.NamedTemporaryFile(mode='w', suffix='.json',
++                                                dir=activity.get_activity_root())
+         json.dump(chart_params, data_file.file)
+         data_file.file.close()
+-        journal_entry.file_path = data_file.name
+-
++        chooser = DesktopSaveChooser(self, filename=data_file.name)
++        filename = chooser.get_filename()
++        if filename:
++            shutil.copyfile(data_file.name,filename)
+         logging.debug('Create %s data file', data_file.name)
+-        datastore.write(journal_entry)
+-        self._show_journal_alert(
+-            _('Exported data'), _('Open in the Journal'),
+-            journal_entry.object_id)
++
```

Comment:
Also, we need to the same trick:

![Screenshot from 2020-01-03 19-15-02](https://user-images.githubusercontent.com/481418/71752397-a8f9c980-2e5d-11ea-8494-3bbb7560eaf1.png)